### PR TITLE
Add Vxlan support

### PR DESCRIPTION
**Pull request by Aegle1**
RFC 7348 - Virtual eXtensible Local Area Network (VXLAN)
 - Add vxlan packet parser
 - Configurable vxlan port is supported
 - Test cases
 - Using different vxlan ports in multithreading scenario needs to be careful. This is the nature when parsers are not containerized.

### CODE REVIEW

File: PacketDotNet/VxlanFields.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        public static readonly int Reserved2Length = 1;
+
+        /// <summary>Length of the vxlan header in bytes.</summary>
+        public static readonly int HeaderLength = FlagsLength + Reserved1Length + VniLength + Reserved2Length;
+
+        /// <summary>IANA assigned Vxlan destination UDP port by default.</summary>
+        public static readonly ushort DefaultUdpDstPortForVxlan = 4789;
+
+        /// <summary>The Vxlan port should be configurable according to RFC 7348.</summary>
+        public static ushort UdpDstPortForVxlan = DefaultUdpDstPortForVxlan;
```
Comment (PhyxionNL) [inline PacketDotNet/VxlanFields.cs:42]: This seems useless.

File: PacketDotNet/VxlanFields.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        public static readonly int VniPosition = 4;
+
+        /// <summary>Length of the Reserved in bytes.</summary>
+        public static readonly int Reserved2Length = 1;
+
+        /// <summary>Length of the vxlan header in bytes.</summary>
+        public static readonly int HeaderLength = FlagsLength + Reserved1Length + VniLength + Reserved2Length;
+
+        /// <summary>IANA assigned Vxlan destination UDP port by default.</summary>
+        public static readonly ushort DefaultUdpDstPortForVxlan = 4789;
```
Comment (PhyxionNL) [inline PacketDotNet/VxlanFields.cs:39]: Rename: DestinationPort.

File: PacketDotNet/VxlanPacket.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+using PacketDotNet.Utils.Converters;
+
+namespace PacketDotNet
+{
+    /// <summary>
+    /// Vxlan
+    /// </summary>
+    public sealed class VxlanPacket : Packet
+    {
+        /// <summary>
```
Comment (PhyxionNL) [inline PacketDotNet/VxlanPacket.cs:22]: Just commenting on this file in general, can you reformat? There's some additional whitespace (two white lines) and also the XML comments are not according to code style, e.g. the values should be on one line where possible, see here: https://github.com/dotpcap/packetnet/blob/af442b608f72d0c3f19ed4a9dc32f950558fde48/PacketDotNet/DhcpV4Packet.cs#L26

File: PacketDotNet/VxlanPacket.cs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        /// Constructor
```
Comment (PhyxionNL) [inline PacketDotNet/VxlanPacket.cs:23]: Initializes a new instance ... ,  see Dhcp link above. Also ctor below.

File: PacketDotNet/VxlanPacket.cs [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        public VxlanPacket(ByteArraySegment byteArraySegment, Packet parentPacket)
+            : this(byteArraySegment)
+        {
+            ParentPacket = parentPacket;
+        }
+
+        /// <summary>
+        /// A valid vxlan flag byte should always be 0x08.
+        /// </summary>
+        public byte Flags => Header.Bytes[Header.Offset + VxlanFields.FlagsPosition];
```
Comment (PhyxionNL) [inline PacketDotNet/VxlanPacket.cs:58]: Please add a set as well.

File: PacketDotNet/VxlanPacket.cs [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        /// A <see cref="Packet" />
+        /// </param>
+        public VxlanPacket(ByteArraySegment byteArraySegment, Packet parentPacket)
+            : this(byteArraySegment)
+        {
+            ParentPacket = parentPacket;
+        }
+
+        /// <summary>
+        /// A valid vxlan flag byte should always be 0x08.
```
Comment (PhyxionNL) [inline PacketDotNet/VxlanPacket.cs:56]: Gets or sets a valid ...

File: PacketDotNet/VxlanPacket.cs [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        /// </summary>
+        public byte Flags => Header.Bytes[Header.Offset + VxlanFields.FlagsPosition];
+
+        /// <summary>
+        /// VXLAN Network Identifier (VNI)
```
Comment (PhyxionNL) [inline PacketDotNet/VxlanPacket.cs:61]: Gets or sets the VXLAN...

**Review by PhyxionNL — CHANGES_REQUESTED** [review]
Few small things, rest looks good, thanks 👍

File: PacketDotNet/VxlanFields.cs (hunk 1 above)
Comment (Aegle1) [reply PacketDotNet/VxlanFields.cs]: Actually custom port is very common in industrial scenarios. Especially for early systems that are still running. I suggest to keep it, as it says in standard: https://www.rfc-editor.org/rfc/rfc7348.html#page-10:~:text=Destination%20Port%3A%20IANA,SHOULD%20be%20configurable.

File: PacketDotNet/VxlanPacket.cs (hunk 3 above)
Comment (Aegle1) [reply PacketDotNet/VxlanPacket.cs:22]: Is coding style with or without new line at EOF? You added new line to GreNvgrePacketTest.cs but DhcpV4Packet.cs doesn't have it.

Comment (PhyxionNL) [reply PacketDotNet/VxlanPacket.cs:22]: I edited it through GitHub, I think that added the character. It doesn't really matter as long as the rest of the class is OK, it seems now 👍

File: PacketDotNet/VxlanFields.cs (hunk 1 above)
Comment (PhyxionNL) [reply PacketDotNet/VxlanFields.cs]: OK, I thought it referred to the default one initially, it's fine then :)

**Pull request merged**